qcengine/programs/molpro.py

```diff
@@ -329,37 +375,64 @@ def parse_output(self, outfiles: Dict[str, str], input_model: 'ResultInput') ->
             del properties['ccsd_singlet_pair_energy']
             del properties['ccsd_triplet_pair_energy']
 
+        # Process data in molpro_variable_map
+        # Note: For the DFT case molecule_method is the name of the functional plus R or U in front
+        molecule = root.find('molpro_uri:job/molpro_uri:molecule', name_space)
+        molecule_method = molecule.attrib['method']
+        molecule_final_energy = float(molecule.attrib['energy'])  # Energy from the molecule tag in case its needed
+        # Loop over each variable under the variables tag to grab additional info from molpro_variable_map
+        for variable in molecule.findall('molpro_uri:variables/molpro_uri:variable', name_space):
```

Comment:
Bleh, XML fun. I don't see a way to make this simpler however.

Reply:
Yes I agree, it's starting to feel somewhat unwieldy although I haven't thought of a good way to simplify. In the future we could write separate functions that parse the different sections of the xml (e.g. energies, dipoles, gradient, variables, basis, etc.). That would at least make it more readable.